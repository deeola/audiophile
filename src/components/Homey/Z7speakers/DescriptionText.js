import React from "react";
import { SecondaryButton } from "../../Buttons/Button";
import { DescCont, Title } from "./Z7speakerstyling";

const DescriptionText = ({ title, link }) => {
  return (
    <DescCont>
      <Title data-testid='title'>{title}</Title>
      <SecondaryButton label={"SEE PRODUCT"} link={link} />
    </DescCont>
  );
};

export default DescriptionText;
