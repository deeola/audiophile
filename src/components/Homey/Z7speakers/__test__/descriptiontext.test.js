import ReactDOM from "react-dom";
import DescriptionText from "../DescriptionText";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

it("renders correctly in dom", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <BrowserRouter>
      <DescriptionText />
    </BrowserRouter>, div
  );
});

it("display text correctly", () => {
    const {getByTestId}  =render(

        <BrowserRouter>
      <DescriptionText title={'hi'} />
    </BrowserRouter>
    )
  
    expect(getByTestId('title')).toHaveTextContent('hi')
    
  });
  
//create snapshot

it('matches snapshot', () => {
    const tree = renderer.create(
        <BrowserRouter>
        <DescriptionText title={'hi'} />
      </BrowserRouter>
    ).toJSON()

    expect(tree).toMatchSnapshot();
})