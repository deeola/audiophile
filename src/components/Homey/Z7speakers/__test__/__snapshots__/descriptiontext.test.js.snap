// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-dkzDqf"
>
  <h4
    className="sc-hKMtZM cXPvZu"
    data-testid="title"
  >
    hi
  </h4>
  <a
    className="sc-bczRLJ cwyJdT"
    color="secondary"
    hover="secondary"
    href=""
    onClick={[Function]}
  >
    SEE PRODUCT
  </a>
</div>
`;
